cumulusci/cli/cci.py

```diff
@@ -1425,83 +1425,235 @@ def task_info(runtime, task_name):
 
-            pdb.set_trace()
-    finally:
-        runtime.alert(f"Task complete: {task_name}")
+    def _convert_old_option_syntax(self, args):
+        """
+        Given a list of args, convert any options specified
+        with the old format `-o name value` to the new format `--name value`.
+
+        Args:
+            param1: The list of arguments to convert
+
+        Returns:
+            The list of arguments
+
+        Raises:
+            CumulusCIUsageError: if there is an error in parsing the old option syntax,
+            or the option doesn't exist for the given task.
+        """
+        duplicate = self._has_duplicate_options(args)
+        if duplicate:
+            raise CumulusCIUsageError(
+                f"Found duplicate option `{duplicate}` in given command:\n cci task run {' '.join(args)}"
+            )
+
+        task_name = args[0]
+        while "-o" in args:
+            idx = args.index("-o")
+            opt_name = args[idx + 1]
+            opt_value = args[idx + 2]
+
+            if opt_name.startswith("-") or opt_value.startswith("-"):
+                raise CumulusCIUsageError(
+                    f"Names and values for options specified with `-o` cannot start with '-'.\nFound: -o {opt_name} {opt_value}"
+                )
+            elif not self._option_in_task(opt_name, task_name):
+                raise CumulusCIUsageError(
+                    f"Task <{task_name}> received unrecognized option: {opt_name}"
+                )
+
+            args[idx + 1] = f"--{opt_name}"
+            args.remove("-o")
+
+        return args
+
+    def _has_duplicate_options(self, args):
+        """Checks for the precense of duplicate options.
+        Returns the first occurrence of a duplicate option
+        name, None otherwise."""
+        opt_names = sorted(self._parse_option_names(args))
+        for i, name in enumerate(opt_names):
+            if i == len(opt_names) - 1:
+                break
+            elif name == opt_names[i + 1]:
+                return name
+
+        return None
+
+    def _parse_option_names(self, args):
+        """
+        Given a list of arguments, parse
+        out the provided option names.
+        new syntax: --name value
+        old syntax: -o name value
+        """
+        opt_names = []
+        for i, opt in enumerate(args):
+            if opt == "-o":
+                opt_names.append(args[i + 1])
+            elif opt.startswith("--"):
+                opt_names.append(opt.strip("-"))
+
+        return opt_names
+
+    def _option_in_task(self, opt_name, task_name):
+        """
+        Returns True if opt_name is the name of an
+        option defined in the given task, else False.
+        """
+        task = RUNTIME.project_config.get_task(task_name)
+        task_class = import_global(task.config["class_path"])
+
+        if "task_options" not in task_class.__dict__.keys():
+            task_option_names = self._get_task_options_in_hierarchy(task_class)
```

Comment:
You don't have to do this. `task_class.task_options` will inherit it from the task class's base classes if it's not defined on task_class itself.